src/OrchardCore/OrchardCore.Users.Core/Services/UserStore.cs

```diff
@@ -110,13 +120,8 @@ public async Task<IdentityResult> DeleteAsync(IUser user, CancellationToken canc
 
         public async Task<IUser> FindByIdAsync(string userId, CancellationToken cancellationToken = default(CancellationToken))
         {
-            int id;
-            if (!int.TryParse(userId, out id))
-            {
-                return null;
-            }
-
-            return await _session.GetAsync<User>(id);
+            // For backwards compatability we check also check the UserId against the NormalizedUserName.
+            return await _session.Query<User, UserIndex>(u => u.UserId == userId || u.NormalizedUserName == userId).FirstOrDefaultAsync();
```

Comment:
For when we review this agin, this is one bit which concerns me, but I think it is ok